app/src/main/java/ch/epfl/cs311/wanderwave/ui/screens/MapScreen.kt

```diff
@@ -1,13 +1,72 @@
 package ch.epfl.cs311.wanderwave.ui.screens
 
+import android.content.ContentValues
+import android.util.Log
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.platform.testTag
 import androidx.compose.ui.tooling.preview.Preview
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import ch.epfl.cs311.wanderwave.model.data.Beacon
+import ch.epfl.cs311.wanderwave.model.local.BeaconEntity
+import ch.epfl.cs311.wanderwave.viewmodel.MapViewModel
+import com.google.firebase.firestore.ktx.firestore
+import com.google.firebase.ktx.Firebase
 import com.google.maps.android.compose.GoogleMap
+import com.google.maps.android.compose.Marker
+import com.google.maps.android.compose.MarkerState
 
 @Composable
 @Preview
 fun MapScreen() {
-  GoogleMap(modifier = Modifier.testTag("mapScreen"))
+  val viewModel: MapViewModel = hiltViewModel()
+  val uiState by viewModel.uiState.collectAsStateWithLifecycle()
+  // addMockBeacons()
+  GoogleMap(modifier = Modifier.testTag("mapScreen")) { DisplayBeacons(uiState.beacons) }
+}
+
+/**
+ * This is a Composable function that displays the beacons on the map. It takes a list of beacons as
+ * input and adds a marker for each beacon on the map.
+ *
+ * @param beacons The list of beacons to be displayed on the map.
+ */
+@Composable
+fun DisplayBeacons(beacons: List<Beacon>) {
+
+  // Add a marker for each beacon
+  // source for the icon: https://www.svgrepo.com/svg/448258/waypoint
+  // val customIcon = BitmapDescriptorFactory.fromResource(R.drawable.waypoint)
+  beacons.forEach() {
+    Marker(
+        state = MarkerState(position = it.location.toLatLng()),
+        title = it.id,
+        // icon = customIcon,
+    )
+  }
+}
+
+fun addMockBeacons() {
+  val db = Firebase.firestore
+  val collection = db.collection("beacons")
+
+  val latitudes = listOf(46.51857556996283, 46.51857417773428, 46.52298529087412, 46.51846723837138)
```

Comment:
Here, maybe instead of having 3 list, we can have only 1 list of data class beacon?(But then I understand it's a mock, so it might bot be how you want to implement the final version)

Reply:
I didn't make this haha, but I will change it